Specs/Scene/ModelExperimental/MorphTargetsPipelineStageSpec.js

```diff
@@ -0,0 +1,238 @@
+
+      MorphTargetsPipelineStage.process(renderResources, primitive);
+
+      const shaderBuilder = renderResources.shaderBuilder;
+      const attributes = renderResources.attributes;
+
+      const length = attributes.length;
+      expect(length).toEqual(6);
+
+      // These values happen to be the same for all attributes
+      const expectedByteOffset = 0;
+      const expectedStride = 12;
+      for (let i = 0; i < length; i++) {
+        const attribute = attributes[i];
+        verifyMorphTargetAttribute(
+          attribute,
+          i + 1,
+          expectedByteOffset,
+          expectedStride
+        );
+      }
+
+      ShaderBuilderTester.expectHasVertexFunction(
+        shaderBuilder,
+        MorphTargetsPipelineStage.FUNCTION_ID_GET_MORPHED_POSITION,
+        MorphTargetsPipelineStage.FUNCTION_SIGNATURE_GET_MORPHED_POSITION,
+        [
+          "    vec3 morphedPosition = position;",
+          "    morphedPosition += u_morphWeights[0] * a_targetPosition_0;",
+          "    morphedPosition += u_morphWeights[1] * a_targetPosition_1;",
+          "    return morphedPosition;",
+        ]
+      );
+
+      ShaderBuilderTester.expectHasVertexFunction(
+        shaderBuilder,
+        MorphTargetsPipelineStage.FUNCTION_ID_GET_MORPHED_NORMAL,
+        MorphTargetsPipelineStage.FUNCTION_SIGNATURE_GET_MORPHED_NORMAL,
+        [
+          "    vec3 morphedNormal = normal;",
+          "    morphedNormal += u_morphWeights[0] * a_targetNormal_0;",
+          "    morphedNormal += u_morphWeights[1] * a_targetNormal_1;",
+          "    return morphedNormal;",
+        ]
+      );
+
+      ShaderBuilderTester.expectHasVertexFunction(
+        shaderBuilder,
+        MorphTargetsPipelineStage.FUNCTION_ID_GET_MORPHED_TANGENT,
+        MorphTargetsPipelineStage.FUNCTION_SIGNATURE_GET_MORPHED_TANGENT,
+        [
+          "    vec3 morphedTangent = tangent;",
+          "    morphedTangent += u_morphWeights[0] * a_targetTangent_0;",
+          "    morphedTangent += u_morphWeights[1] * a_targetTangent_1;",
+          "    return morphedTangent;",
+        ]
+      );
+
+      ShaderBuilderTester.expectHasVertexDefines(shaderBuilder, [
+        "HAS_MORPH_TARGETS",
+      ]);
+
+      ShaderBuilderTester.expectHasAttributes(shaderBuilder, undefined, [
+        "attribute vec3 a_targetPosition_0;",
+        "attribute vec3 a_targetPosition_1;",
+        "attribute vec3 a_targetNormal_0;",
+        "attribute vec3 a_targetNormal_1;",
+        "attribute vec3 a_targetTangent_0;",
+        "attribute vec3 a_targetTangent_1;",
+      ]);
+
+      ShaderBuilderTester.expectVertexLinesEqual(shaderBuilder, [
+        _shadersMorphTargetsStageVS,
+      ]);
+
+      ShaderBuilderTester.expectHasVertexUniforms(shaderBuilder, [
+        "uniform float u_morphWeights[2];",
+      ]);
+
+      const uniformMap = renderResources.uniformMap;
+      expect(uniformMap.u_morphWeights()).toBe(primitive.morphWeights);
+    });
+  });
+});
```

Comment:
Oh, since this uses GltfLoader (which may create GPU resources along the way), we should mark this as a WebGL test.
```suggestion
}, "WebGL");
```

We really should do a sweep through the pipeline stages at some point to see what test suites might be missing this. Not everyone needs it, but some do.